### ISSUE TITLE
Don't apply video compression to files

### ISSUE THREAD

**Issue by licaon-kter** [siacs/Conversations#2644]
#### General information

* **Version:** 1.21.0
* **Conversations source:** F-Droid

#### Steps to reproduce

Attach a video

#### Expected result

Video is send as a file.

#### Actual result

Video file is treated as an "in app recorded video file" and recompressed, which in certain conditions, might yield a file bigger than the original.

Either add a Setting like we have for pictures (auto/never/compress) or maybe be smart about it and if the compressed output is bigger than the original just use the original.

#### Debug output

to come

**Comment by annafallasktick** [comment]
I, too, am not a fan of recoding video in the same way I am not a fan of jpeg-rape iterations.

**Comment by licaon-kter** [comment]
Can we reopen this?

The integrated compression, while fast, yields pretty big files and we can't control its parameters.

So I use https://github.com/brarcher/video-transcoder to make the smaller (just reducing bitrate to 2048 yields small files but pretty ok to be viewed on a phone) eg. 226Mb video was compressed to 76Mb in Conversations but 27Mb in transcoder. 

Ok, but...while I waited a bit longer for transcoder to yield that nice small 27Mb file....when I _attach file_ Conversations will try (and fail) to make it smaller, making the user wait yet again for something that's not needed.

**Comment by licaon-kter** [comment]
Another bump here, pls reopen?

Currently I can set pictures to compress if bigger, but use "attach file" to use the original file so I'm not forced to change the in-app settings every time I want the picture to look ok and not blurred with scaling _(also [SendReduced](https://f-droid.org/en/packages/mobi.omegacentauri.SendReduced/) appears to not work on Android 10 anymore unfortunately :()_.

Buuut, I can't do this with video...it's either "uncompressed" always or "forced recompress" :(

**Comment by aslmx** [comment]
Came accross this as well. Using Proxy65 for big files anyway... so a way to send original size videos would be appreciated.

Please reopen.

**Comment by iNPUTmice** [comment]
@aslmx you can disable video compression in the settings.

**Comment by aslmx** [comment]
> @aslmx you can disable video compression in the settings.

I don't want to disable it in general.

If i attach a "file" i want the file to be transmitted unaltered. If i attach a "photo" - "photo" being a synonym for "media" - i'm fine with compression....

My usecase is sending a video (via Proxy65) in full original quality as a "file".  (And even ifnot using Proxy65 someone might want to send the original quality.

If i select photos and send them as a "file", they won't get compressed either.

**TL;DR: In my opinion it would be more consistent to compress (if enabled) Videos and Photo/Pictures if send as Picture(/Media) and not apply compression if anything is sent as "file"**

Do you agree?

**Comment by paskalito** [comment]
I Fully agree, and want to add, that if i select a file to be sent to someone i expect it to arrive exactly the way i sent it. Altering a "File" seems to be Wrong.
